pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -615,7 +620,44 @@ private static Expression toExpression(SqlNode node) {
             funcExpr.getFunctionCall().addToOperands(toExpression(child));
           }
         }
+
+        if (FunctionRegistry.containsFunctionByName(funcName) && isCompileTimeEvaluationPossible(funcExpr)) {
+          int functionOperandsLength = funcSqlNode.getOperands().length;
+          FunctionInfo functionInfo = FunctionRegistry.getFunctionByName(funcName);
+          Object[] arguments = new Object[functionOperandsLength];
+          for (int i = 0; i < functionOperandsLength; i++) {
+            SqlLiteral argSqlNode = (SqlLiteral) funcSqlNode.getOperands()[i];
+            arguments[i] = argSqlNode.toValue();
+          }
+          try {
+            FunctionInvoker invoker = new FunctionInvoker(functionInfo);
+            funcExpr = RequestUtils.getLiteralExpression(invoker.process(arguments).toString());
+          } catch (Exception e) {
+            throw new SqlCompilationException("Unsupported Scalar function - " + funcName);
+          }
+        }
         return funcExpr;
     }
   }
+
+  /**
+   * Utility method to check if the function can be evaluated during the query compilation phae
+   * @param funcExpr
+   * @return true if all arguments are literals
+   */
+  private static boolean isCompileTimeEvaluationPossible(Expression funcExpr) {
```

Comment:
Instead of this, why don't we just have a constant value function registry  where all compile time evaluated functions are registered. So this check then becomes if the function is part of the constant value function registry.

Reply:
One thing here is that the function could be used in transform functions in query/ingestion field conversion. Ideally we should be able to evaluate any transform function with literal here.

Reply:
+1